### DISCUSSION TITLE
Duck cli failing to authenticate to onedrive

### DISCUSSION THREAD

**Discussion by enthusedcoder** [iterate-ch/cyberduck#16199] [Q&A]
Hello everyone.  So, I have been a long time user of cyberduck, and I love the product.  However, I need to use the duck cli for a little project I am working on, and I need to get it to connect to onedrive.  So, I install the duck cli, and then I run the following command in my command prompt:

```
duck -l onedrive:/
```

The duck cli application then launches a web browser so that I can authenticate to onedrive via the oauth flow.  So, I plug in my credentials and log into onedrive.  At this point, one of two scenarios will occur, and it is dependent on whether or not the computer that I am running the duck cli has cyberduck installed.  If the computer has cyberduck installed, the browser will ask for permission to open the cyberduck application, which is odd, but ok, I will let the browser launch cyberduck.  Maybe cyberduck acts as the middle man between the browser and the cli application and just passes the authentication along.  But no,  Cyberduck is launched, and nothing happens.  In either cyberduck, or the command prompt (which still reads "Open web browser to authenticate and obtain an authorization code").  If the computer does not have cyberduck installed on it, then I am presented with this screen indefinitely:

![duck](https://github.com/user-attachments/assets/1940ac77-ad21-4dff-a1d2-231342729c27)

And I am forced to just close the browser window and kill the duck cli process.  Am I doing something wrong here?  I have looked at the [Onedrive Configuration](https://docs.duck.sh/protocols/onedrive/) page for the duck cli, but nothing there indicates that I am doing anything wrong, or any kind of solution.  I have tried several different browsers on several different computers, I have cleared the cookies in those browsers to make sure there isn't any kind of issue with multiple microsoft accounts, and I have come up with nothing.  Is anyone else having this issue?  Any advice as to how I can fix this?  Or do I need to report this as a bug?  Thanks in advance.

**Comment by dkocher** [comment]
This is an open issue [^1] with a [workaround](iterate-ch/cyberduck#13587#issuecomment-1885628198).

[^1]: iterate-ch/cyberduck#13587

**Reply by enthusedcoder** [reply]
Hello sir.  Thank you for your reply.  So, based on the discussion that you linked, I would need to authenticate using cyberduck and then pass the profile to the command line as an argument.  I was actually looking into this, and that brings up another question.  I have cyberduck installed on another machine, and have already authenticated to onedrive with that cyberduck instance.  While reviewing the documentation trying to troubleshoot this problem, I found a file named "Onedrive.duck" in "%APPDATA%\Cyberduck\History" which, after looking at the contents, I assume is the file that cyberduck uses to authenticate to Onedrive.  My question is can I take that file and supply that as the duck cli profile argument?
